src/nxdoc/nuxeo-server/upgrading-the-nuxeo-platform/how-to-upgrade-from-lts-2023-to-lts-2025/how-to-upgrade-search-service.md

```diff
@@ -0,0 +1,301 @@
+    <hint name="foo" class="..." />
+```
+to 
+```xml
+<extension target="org.nuxeo.ecm.core.search.client.opensearch1" point="hint">
+  <hint name="foo" class="..." />
+```
+
+## Automation Operations
+
+`ElasticSearch.Index` should be replaced by `Search.Index`.
+
+`Elasticsearch.BulkIndex` should be replaced by `Search.Index`.
+
+`Elasticsearch.WaitForIndexing` should be replaced by `Search.WaitForIndexing`.
+
+## Management API
+
+`/management/elasticsearch` call should be changed to `/management/search` and stay compatible.
+
+There is no more low level `/management/elasticsearch/flush` and `/management/elasticsearch/optimize`. We recommend to use directly the search engine to perform such administrative operation.
+
+
+## Nuxeo Server Configuration
+
+{{#> callout type='warning' }}
+Nuxeo Server is provided with a `repository` search client implementation.
+You need to explicitly install a search client Marketplace Package to have an `enhanced` index.
+{{/callout}}
+
+### OpenSearch Search Client
+
+To use OpenSearch indexing and search capability you need to install the [nuxeo-search-client-opensearch1](https://connect.nuxeo.com/nuxeo/site/marketplace/package/nuxeo-search-client-opensearch1)
+package.
+
+After installation, check that the `opensearch1-search-client` template is included in your `nuxeo.conf` file. The template will enable to access an OpenSearch 1.x, Elasticsearch 7.x or 8.x cluster. The previous `elasticsearch.enabled` configuration property has no more effect.
+
+Previously, the ElasticSearch / OpenSearch configuration properties looked like:
+
+```properties
+# generic
+elasticsearch.addressList=http://localhost:9200
+elasticsearch.index.translog.durability=request
+elasticsearch.indexNumberOfShards=5
+elasticsearch.indexNumberOfReplicas=1
+elasticsearch.restClient.connectionTimeoutMs=30000
+elasticsearch.restClient.socketTimeoutMs=121000
+elasticsearch.restClient.ssl.certificate.verification=true
+
+# optional
+elasticsearch.restClient.username=
+elasticsearch.restClient.password=
+elasticsearch.restClient.truststore.path=
+elasticsearch.restClient.truststore.password=
+elasticsearch.restClient.truststore.type=
+elasticsearch.restClient.keystore.path=
+elasticsearch.restClient.keystore.password=
+elasticsearch.restClient.keystore.type=
+```
+
+Now you will configure OpenSearch 1 / ElasticSearch 7 or 8 connection with:
+
+```properties
+# generic
+nuxeo.opensearch1.client.server=http://localhost:9200
+
+nuxeo.opensearch1.client.connectionTimeout=3s
+nuxeo.opensearch1.client.socketTimeout=121s
+nuxeo.opensearch1.client.sslCertificateVerification=true
+
+# optional
+nuxeo.opensearch1.client.username=
+nuxeo.opensearch1.client.password=
+nuxeo.opensearch1.client.trustStore.path=
+nuxeo.opensearch1.client.trustStore.password=
+nuxeo.opensearch1.client.trustStore.type=
+nuxeo.opensearch1.client.keyStore.path=
+nuxeo.opensearch1.client.keyStore.password=
+nuxeo.opensearch1.client.keyStore.type=
+```
```

Comment:
What about adding translog durability, indexNumberOfShards and indexNumberOfReplicas as they are mentioned in the previous part about configuration parameter for LTS 2023?

Reply:
Also, should we mention `nuxeo.search.client.default.opensearch1.index.name`?